src/sentry/sentry_metrics/querying/data/execution.py

```diff
@@ -89,23 +89,27 @@ def _push_down_group_filters(
     if not groups_collection:
         return metrics_query
 
-    # We perform a transformation in the form [(key_1 = value_1 AND key_2 = value_2) OR (key_3 = value_3)].
-    groups_filters = []
+    groups_keys = set()
     for groups in groups_collection:
-        inner_snuba_filters = []
-        for filter_key, filter_value in groups:
-            inner_snuba_filters.append(Condition(Column(filter_key), Op.EQ, filter_value))
-
-        # In case we have more than one filter, we have to group them into an `AND`.
-        if len(inner_snuba_filters) > 1:
-            groups_filters.append(BooleanCondition(BooleanOp.AND, inner_snuba_filters))
-        else:
-            groups_filters.append(inner_snuba_filters[0])
-
-    # In case we have more than one filter, we have to group them into an `OR`.
-    if len(groups_filters) > 1:
-        groups_filters = [BooleanCondition(BooleanOp.OR, groups_filters)]
+        for filter_key, _ in groups:
+            groups_keys.add(filter_key)
 
+    groups_keys = sorted(list(groups_keys))
+    groups_values_tuples = []
+    for groups in groups_collection:
+        group_values = []
+        for group_key in groups_keys:
+            for filter_key, filter_value in groups:
```

Comment:
Added a nested loop here since it's not really needed to create a reverse index to make the lookup faster. We expect that users will group by a very small number of groups.